### ISSUE TITLE
error attaching a file undefined500

### ISSUE THREAD

**Issue by shields1968** [osTicket/osTicket#6337]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ ] Do you have any errors in the PHP error log, or javascript console?
* [X] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [ ] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description

[Description of the bug or feature]

the problem I have is that I cannot attach files when a user creates a new ticket
also do not paste an image inside the ticket
however it was working fine before 

Expected behavior:

files to be attached

Actual behavior:

it does not attach the files nor can you drag an image
when creating a ticket

I attach an image of the error

![error2](https://user-images.githubusercontent.com/71939603/198850466-0451a745-4d2d-4088-a45c-e7e6fca0c4a8.png)

 Versions

server information
--
Versión osTicket | v1.15.8 (0daca43) —  Hasta la fecha
Servidor del Software Web | Apache
Versión MySQL | 5.7.40
Versión PHP | 7.3.33

**Comment by JediKev** [comment]
@shields1968 

Did you try the below?
- https://github.com/osTicket/osTicket/pull/6308

If that works then you need to go to **Admin Panel > Settings > Tickets**, click **Config** next to **Ticket Attachment Settings**, configure **Maximum File Size**, and click **Save**.

Cheers.

**Comment by shields1968** [comment]
It is already configured at 128 mb and as I said it worked fine.
but now not even if they are files smaller than configured

**Comment by JediKev** [comment]
@shields1968 

Can you open the browser's Developer Tools (right click the page and click Inspect), go to the Network tab, upload a file, click on the failed request in the Network tab, and click the Response tab within that. There you should see a more detailed error message.

Cheers.

**Comment by shields1968** [comment]
Nothing comes out or I don't know how to do it
I send you the link so you can see what happens

**Comment by shields1968** [comment]
Response 
Unable to store file: File upload error #6

**Comment by JediKev** [comment]
@shields1968 

File Upload Error 6 means missing temporary folder. Maybe you have path issues, permission issues, etc. but the below link should help you.
- https://stackoverflow.com/questions/27099537/upload-file-to-web-server-error-6-php

Cheers.

**Comment by shields1968** [comment]
that comes out

![error 3](https://user-images.githubusercontent.com/71939603/198853959-af7502e8-6765-423c-9dd9-407be364cc0e.png)

**Comment by shields1968** [comment]
it may be a php problem
or from the server, because it is on a shared server

**Comment by JediKev** [comment]
@shields1968 

Then I would definitely recommend contacting your hosting provider for further assistance.

Cheers.

**Comment by shields1968** [comment]
ok 

Thank you

**Comment by RikkiMasters** [comment]
Same issue here. @shields1968 did you ever fix?

Version 1.16.1. No errors in logs or response. Will be updating to latest version today.

**Comment by lourdas** [comment]
Same issue here, using version 1.17.3. No error in apache or php fpm logs.

**Comment by lourdas** [comment]
I found this: https://docs.sportdata.org/knowledge-base/fix-file-upload-error-http-500-undefined/ which fixes the issue for my 1.17.3 installation. Tried with zip, 7z or rar files and it works.

**Comment by F5Brewery** [comment]
I updated from 1.17.2 -> 1.17.3, had odd issues in uploading 1.17.2, but none the less it would eventually work.

Did all the actions listed above.  I am now able to upload certain files (was not initially) - zip, txt, pdf for example.  Many others I am not.  I have no restrictions.  Things like tgz, exe, readme.md, tons of others end up with the.
One of my users can not upload a zip.

error attaching a file undefined 500

**Comment by F5Brewery** [comment]
Just to update my previous comment.  So far only I am able to upload Zip files.  2 of my users, the only 2 I have tested, are not able to upload a zip file.  They are able to upload PDF and TXT however.  With the zip file in question I renamed to be a PDF and still encountered the same error.  
I made one of the users an admin in hopes...but it still did not work.
I also cleared temp from browser cache for both users (I did not have to do this for me).

**Comment by JediKev** [comment]
@F5Brewery 

Just comment out the following lines:
- https://github.com/osTicket/osTicket/blob/develop/include/class.forms.php#L3974-L3977

However, _**apply these changes at your own risk**_. Doing so will open your instance to possible file upload vulns. With this being said, we are looking into an alternative MIME type check, if possible.

Cheers.

**Comment by F5Brewery** [comment]
Thank you for this JediKev and the prompt response.  I have performed the edit as prescribed.  This did work!  I do understand the risks, hopefully a good fix will be posted back to here as well.
I am unclear why it would work for me and not the user when I did the same file and also why it took effect only after the upgrade to 1.17.3.  Anyhow it keeps me working and I appreciate this very much.

**Comment by ldev-coder** [comment]
Hi, do you have alternatives?
